src/lib.rs

```diff
@@ -174,26 +174,7 @@ impl<T: Publisher + Clone + 'static> DeviceManager<T> {
         let publisher = self.publisher.clone();
         tokio::spawn(async move {
             while let Some(msg) = telemetry_rx.recv().await {
-                match msg.payload {
-                    TelemetryPayload::SystemStatus(data) => {
-                        let _ = publisher
-                            .send_object(
-                                "io.edgehog.devicemanager.SystemStatus",
-                                "/systemStatus",
-                                data,
-                            )
-                            .await;
-                    }
-                    TelemetryPayload::StorageUsage(data) => {
-                        let _ = publisher
-                            .send_object(
-                                "io.edgehog.devicemanager.StorageUsage",
-                                format!("/{}", msg.path).as_str(),
-                                data,
-                            )
-                            .await;
-                    }
-                };
+                send_telemetry(Box::new(&publisher), msg).await;
```

Comment:
There's no need to allocate on the heap to pass the publisher reference:
- Move `send_telemetry` _inside_ the `DeviceManager` struct so you can use `T` and change its signature to:
```rust
    async fn send_telemetry(publisher: &T, msg: TelemetryMessage)
```
- At the callsite:
```rust
    Self::send_telemetry(&publisher, msg).await;
```